lib/src/domain/model/article.dart

```diff
@@ -1,17 +1,42 @@
-import 'package:json_annotation/json_annotation.dart';
+import 'package:allthenews/src/data/response/article_response.dart';
 
-part 'article.g.dart';
-
-@JsonSerializable()
 class Article {
   final int id;
 
+  final String url;
+
+  final String updated;
+
+  final String authorName;
+
   final String title;
 
-  Article(this.id, this.title);
+  final String abstract;
+
+  final String thumbnail;
+
+  Article({
+    this.id,
+    this.url,
+    this.updated,
+    this.authorName,
+    this.title,
+    this.abstract,
+    this.thumbnail,
+  });
 
-  factory Article.fromJson(Map<String, dynamic> json) => _$ArticleFromJson(json);
+  factory Article.fromResponse(ArticleResponse response) => Article(
```

Comment:
żeby było zgodnie ze sztuką to domena nie powinna zależeć od warstwy data, czyli albo w klasie ArticleResponse zrobiłbym metodę `toArticle` albo taka metoda jak tu czyli Article.fromReponse ale zadeklarowana jako extension w pakiecie data :)